tests/global_communication/test_communicator.cpp

```diff
@@ -103,10 +103,16 @@ TEST(communicator, gather_spikes_equal) {
     // is a list of num_domains*n_local_spikes spikes that have
     // contiguous source gid
     const auto& spikes = global_spikes.values();
+    EXPECT_EQ(n_local_spikes*policy::size(), int(spikes.size()));
     for (auto i=0u; i<spikes.size(); ++i) {
         const auto s = spikes[i];
         EXPECT_EQ(i, unsigned(s.source.gid));
-        EXPECT_EQ(0, s.domain);
+        if (is_dry_run()) {
```

Comment:
I guess I still don't understand why the dry-run-mode sources will always be domain 0. If dry run mode ranks > 1, then the sources could be from any of those ranks and thus the domains could be anywhere (and that's what I've seen debugging)